### PR TITLE
Fix findLatest throws NoSuchFileException with non-existing directory

### PR DESCRIPTION
**Pull request by msengbusch**
### Description

In DateTimeSegment of V3.0 branch findLatest is throwing a NoSuchFileException if parentDirectory does not exist. This occurs when a FileWriter writes to a log file whose parent directory does not exist yet for example on the first run.

### Definition of Done

- [x] I read [contributing.md](https://github.com/tinylog-org/tinylog/blob/v2.4/contributing.md)
- [x] There are no TODOs left in the code
- [x] Code style follows the tinylog standard
- [x] All classes and methods have Javadoc
- [x] Changes are covered by JUnit tests including edge cases, errors, and exception handling
- [x] Maven build works including compiling, tests, and checks (`mvn verify`)
- [x] Changes are committed by a verified email address that is assigned to the GitHub account (https://github.com/settings/emails)

### Agreements

- [x] I agree that my changes will be published under the terms of the [Apache License 2.0](https://github.com/tinylog-org/tinylog/blob/v2.0/license.txt) (mandatory)
- [x] I agree that my GitHub user name will be published in the release notes (optional)

### CODE REVIEW

**Comment by pmwmedia** [comment]
Thank you for the first pull request for tinylog 3 :)

File: tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java [hunk 1]
```diff
@@ -59,6 +59,10 @@ public DateTimeSegment(String pattern, Locale locale) {
 
 	@Override
 	public String findLatest(Path parentDirectory, String prefix) throws IOException {
+		if (!Files.exists(parentDirectory)) {
+			return null;
+		}
```
Comment (pmwmedia) [inline tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java]: Thank you for the fix. I have noticed that we have the same issue in `CountSegment` (and probably other future path segments). What do you think about fixing the issue for all segments in [org/tinylog/impl/path/DynamicPath.java#L40](https://github.com/tinylog-org/tinylog/blob/8f7ffbd34bc45e1fe09c696aad89af041ce9bd92/tinylog-impl/src/main/java/org/tinylog/impl/path/DynamicPath.java#L40)? You would just have to move the new if statement there.

**Review by pmwmedia — CHANGES_REQUESTED** [review]

File: tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java (hunk 1 above)
Comment (msengbusch) [reply tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java]: Yeah of course. I will do it immediately

Comment (pmwmedia) [reply tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java]: I think it should be inside of the loop after `for (int i = 0; i < pathSegments.length; ++i) {` and before `String result = pathSegments[i].findLatest(parentDirectory, prefix);`. So, your check will be executed before calling each path segment.

Could you update the JUnit test, too?

**Comment by pmwmedia** [comment]
Could you add a JUnit test?

File: tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java (hunk 1 above)
Comment (msengbusch) [reply tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java]: I moved it inside the loop and wrote a unit test.

**Review by pmwmedia — APPROVED** [review]

**Comment by pmwmedia** [comment]
Perfect :) Thank you for your contribution!
